### PR TITLE
Add-on store: add warning dialog

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
None

### Summary of the issue:
Users should be informed of the risk of installing add-ons

### Description of user facing changes
When opening the add-on store, a warning dialog is shown.
There is a checkbox to not show the warning again.

### Description of development approach
Uses similar approach to the user stats dialog
Move banner creation to helper function

Added typing to `guiHelper.BoxSizerHelper`. Fix bug where if a `StaticBoxSizer` is used with an intflag of buttons, `BoxSizerHelper.addDialogDismissButtons` wouldn't work correctly

Add support for creating labelled checkboxes using the `BoxSizerHelper`, `LabelledControlHelper` and `associateElements`

### Testing strategy:
Test banner using `--disable-addons`
Test with sample code in source code for `addDialogDismissButtons` bug.
Test the warning nag - ensure the checkbox to hide the warning works

### Known issues with pull request:
None
### Change log entries:
N/A
### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Review by Qchristensen — DISMISSED** [review]
Great work, thanks Sean!

File: source/gui/_addonStoreGui/controls/storeDialog.py [hunk 1]
```diff
@@ -278,6 +294,15 @@ def _channelFilterKey(self) -> Channel:
 	def _titleText(self) -> str:
 		return f"{self.title} - {self._statusFilterKey.displayString} ({self._channelFilterKey.displayString})"
 
+	_warningText = pgettext(
```
Comment (michaelDCurran) [inline source/gui/_addonStoreGui/controls/storeDialog.py:297]: As the warning text is a multiline text control, NVDA only reads the first line when it is focused.
On my system I only hear:
"Add-ons are created by the NVDA community and are not vetted by NV Access. NV Access cannot be held responsible for add-on behavior. The functionality of add-ons is"
Of course I can then press down arrow to hear the rest.
I wonder if we can get away with a much shorter to the point message which can be read in one line such as:
"Add-ons are provided by the community and not vetted or maintained by NV Access. Add-ons could make your system insecure or unstable. Install at your own risk."
This is 157 characters which fits on one line on my system at least.
We could revisit the message when we have more time to come up with a better control which can have the entire message read on focus.

**Comment by XLTechie** [comment]
+1 to @michaelDCurran's proposed revision.

This shouldn't need to be verbose. People know what add-ons are, and if not 
they have the manual. A brief reminder should be all that is needed.

**Comment by XLTechie** [comment]
Actually, @seanbudd, I would prefer to see this as an one-time dialog that comes 
up on store first open, that gives the warning, and then gets far far out of the 
way thereafter. Possibly with a "don't show this again".

Then no in-dialog nag would be required.

Just an alternate opinion.

**Comment by michaelDCurran** [comment]
If it were a one-time dialog, which I would also support, then the 
original message is okay I think.

**Comment by CyrilleB79** [comment]
I also support the one-time dialog. A disclaimer at each use of the add-on store is too invasive IMO. more so when it adds a tab keypress to the navigation flow.

In addition, I think that the disclaimer in the User Guide should be more explicit (and longer than the one of the GUI if needed). A disclaimer paragraph located just after the introduction of the add-on store's global description would be the best IMO, i.e. a 13.1 paragraph inserted after the introduction of section 13 and before the current paragraph 13.1.

At last, if you stick with the disclaimer in the add-on store's GUI:
* do not reduce the message to one line, because you cannot know if the message will fit on one line on all systems, e.g. in case  the message is translated or if the user uses different screen parameters (screen size, font size, resizing the Window, etc.)
* do not use a control that can be focused or consumes a tab keypress while navigating. Rather use something such as panel descriptions in some setting panels such as "Object presentation" or "Document formatting" panels.

**Comment by michaelDCurran** [comment]
I second the comment from @CyrilleB79 
 that it should be an Okay button with a checkbox. However, I think saving in the config is fine (at least for NVDA 2023.2) as other dialogs like the Welcome dialog also save in the config. Worst case is that the user is asked again next time, and has to eventually save their config.

**Review by michaelDCurran — APPROVED** [review]

**Comment by CyrilleB79** [comment]
@seanbudd:
When the warning dialog is open (even in background), we cannot activate NVDA's menu with NVDA+N, but there is not the usual message indicating "Action unavailable while a dialog is expecting an answer".
Could you have a look to it?

**Pull request merged**